File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.cpp
class Solution {
public:
 
    bool searchMatrix(vector<vector<int>>& mat, int target) {
         int row =0, col = mat[0].size()-1,mid=0,n=mat.size()-1;
         while(row <= mat.size()-1 && col >=0){
             
             if(mat[row][col] == target) return true;
             else if(mat[row][col] < target) row++;
             else col--;         
         }
         return false;
    }
};

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {

        
        int row = matrix.length-1, col = 0;

        while(col<=matrix[0].length-1 && row >=0){
           if(matrix[row][col] == target) return true;

           else if(matrix[row][col] > target){
             row--;
           }
           else{
            col++;
           }
        }
        return false;
    }
}